File: webservices/src/main/java/fieg/modulos/cadastro/cliente/dto/SituacaoClienteDTO.java
package fieg.modulos.cadastro.cliente.dto;

import fieg.core.pagination.PageQuery;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.math.BigDecimal;

@Data
@AllArgsConstructor
public class SituacaoClienteDTO extends PageQuery {

    private String statusInterface;
    private Integer idSistema;
    private String clienteDescricao;
    private String cpfCnpj ;
    private String codUnidade;
    private String unidadeDescricao;
    private char entidade;
    private String dtInicioCobranca;
    private String dtFimCobranca;
    private Integer contrato;
    private Integer parcela;
    private BigDecimal vlCobranca;
    private BigDecimal vlPago;
    private BigDecimal vlEstorno;
    private String dtVencimento;
    private String dtPagamento;
    private String dtCredito;
    private String dtEstorno;
    private String dtCancelamento;
    private String cbcSituacao;
    private String objetoContrato;
    private Integer idCobrancaCliente;
    private String nossoNumero ;
    private String rede ;

    private String consumidorNome ;




}

File: web/src/models/DTOs/Cobrancas/SituacaoClienteDTO.ts
import { UnidadeDTO } from "../Outros/UnidadeDTO";
export interface SituacaoClienteComUnidadeDTO {
  situacaoCliente: SituacaoClienteDTO;
  unidade: UnidadeDTO;
}

export interface SituacaoClienteDTO {
  statusInterface: string;
  idSistema: number;
  clienteDescricao: string;
  cpfCnpj: string;
  codUnidade: string;
  unidadeDescricao: string;
  entidade: string;
  dtInicioCobranca: string;
  dtFimCobranca: string;
  contrato: number;
  parcela: number;
  vlCobranca: number;
  vlPago: number | null;
  vlEstorno: number | null;
  dtVencimento: string;
  dtPagamento: string | null;
  dtCredito: string | null;
  dtEstorno: string | null;
  dtCancelamento: string | null;
  cbcSituacao: string;
  objetoContrato: string | null;
  idCobrancaCliente: number;
  nossoNumero: string;
  rede: string;

  consumidorNome: string;
}
